robot_smach_states/src/robot_smach_states/manipulation/grab.py

```diff
@@ -96,7 +99,6 @@ def execute(self, userdata):
         if not arm:
             rospy.logerr("Could not resolve arm")
             return "failed"
-        userdata.arm = arm.side  # Using userdata makes sure we don't need to do any more arm entity magic
```

Comment:
So how do you know that the arm you used for pre-grasping is also the arm used for grasping? This is not checked in any case so you´ve made it much less robust IMHO. If you don't like the user data stuff you must somehow ensure that the designator is locked so that it *can't go wrong*. That does make the designator stuff much more difficult so I don't understand why you reverted this.

Reply:
This was also not the case before.

Reply:
For sure it was for the reset (see changes between lines 290 and 300). If now for a robot with two arms the first arm gives an error in grasping, it might resolve to the second arm and reset that one. You now reverted it to the old situation that has proven to work very poorly... There was even a comment on this in the code....

Reply:
Indeed for the reset it was. But it is better to do the locking.